code/modules/materials/definitions/solids/materials_solid_mineral.dm

```diff
@@ -290,4 +290,236 @@
+	fuel_value = 0.9
+	dirtiness = 15
+	dissolves_into = list(
+		/decl/material/solid/carbon = 0.1,
+		/decl/material/liquid/plasticide = 0.4,
+		/decl/material/liquid/acetone = 0.4,
+		/decl/material/solid/gemstone/diamond = 0.1
+	)
+
+/decl/material/solid/mineral/cassiterite
+	name = "cassiterite"
+	color = "#a1a4cf"
+	heating_products = list(
+		/decl/material/solid/metal/tin = 0.7,
+		/decl/material/solid/metal/tungsten = 0.2,
+		/decl/material/solid/slag = 0.1
+	)
+	ore_name = "cassiterite"
+	ore_result_amount = 10
+	ore_spread_chance = 10
+	ore_scan_icon = "mineral_common"
+	ore_icon_overlay = "lump"
+	ore_type_value = ORE_SURFACE
+	ore_data_value = 1
+	value = 0.9
+	sparse_material_weight = 20
+	rich_material_weight = 10
+	fuel_value = 0.8
+	dissolves_into = list(
+		/decl/material/solid/metal/tin = 0.8,
+		/decl/material/solid/metal/tungsten = 0.2
+	)
+
+/decl/material/solid/mineral/wolframite
+	name = "wolframite"
+	color = "#8184ac"
+	heating_products = list(
+		/decl/material/solid/metal/tin = 0.1,
+		/decl/material/solid/metal/tungsten = 0.2,
+		/decl/material/solid/metal/iron = 0.2,
+		/decl/material/solid/slag = 0.5
+	)
+	ore_name = "wolframite"
+	ore_result_amount = 5
+	ore_spread_chance = 15
+	ore_scan_icon = "mineral_uncommon"
+	ore_icon_overlay = "lump"
+	ore_type_value = ORE_SURFACE
+	ore_data_value = 1
+	value = 0.9
+	sparse_material_weight = 15
+	rich_material_weight = 10
+	fuel_value = 0.8
+	dissolves_into = list(
+		/decl/material/solid/metal/tin = 0.1,
+		/decl/material/solid/metal/tungsten = 0.6,
+		/decl/material/solid/metal/iron = 0.3
+	)
+
+/decl/material/solid/mineral/sperrylite
+	name = "sperrylite"
+	color = "#cfd0d8"
+	heating_products = list(
+		/decl/material/solid/metal/platinum = 0.5,
+		/decl/material/solid/metal/iron = 0.1,
+		/decl/material/solid/glass = 0.1,
+		/decl/material/solid/slag = 0.3
+	)
+	ore_name = "sperrylite"
+	ore_result_amount = 5
+	ore_spread_chance = 15
+	ore_scan_icon = "mineral_uncommon"
+	ore_icon_overlay = "lump"
+	ore_type_value = ORE_PRECIOUS
+	ore_data_value = 1
+	value = 1.1
+	sparse_material_weight = 10
+	rich_material_weight = 5
+	fuel_value = 0.8
+	dissolves_into = list(
+		/decl/material/solid/metal/platinum = 0.7,
+		/decl/material/solid/metal/iron = 0.1,
+		/decl/material/solid/metal/glass = 0.1,
+		/decl/material/solid/metal/titanium = 0.1
+	)
+
+/decl/material/solid/mineral/sphalerite
+	name = "sphalerite"
+	color = "#aaaa9c"
+	heating_products = list(
+		/decl/material/solid/metal/zinc = 0.7,
+		/decl/material/solid/metal/iron = 0.1,
+		/decl/material/solid/slag = 0.2
+	)
+	ore_name = "sphalerite"
+	ore_result_amount = 10
+	ore_spread_chance = 15
+	ore_scan_icon = "mineral_common"
+	ore_icon_overlay = "lump"
+	ore_type_value = ORE_SURFACE
+	ore_data_value = 1
+	value = 0.8
+	sparse_material_weight = 25
+	rich_material_weight = 15
+	fuel_value = 0.8
+	dissolves_into = list(
+		/decl/material/solid/metal/zinc = 0.7,
+		/decl/material/solid/metal/iron = 0.2
```

Comment:
Needs a comma.

Reply:
Resolving all of the above shortly. I called oil shale 'Dense Graphite' before, so I'm gonna rename it to that